lua/entities/gmod_wire_expression2/core/selfaware.lua

```diff
@@ -108,8 +108,9 @@ end
 
 -- Get the name of another E2
 e2function string entity:getName()
-	if (!IsValid(this) or this:GetClass() != "gmod_wire_expression2") then return "" end
-	return this.name or ""
+	if (IsValid(this) and this.GetGateName) then
+		return this:GetGateName()
+	end
```

Comment:
Ohhh I know what you mean now. Isn't it save to assume name isn't nil? I guess I'll add it anyway.